### PR TITLE
Refactor to use dataclasses

### PR DESCRIPTION
**Pull request by MichaelClifford**
This is a WIP PR, where most of the flatliners have been updated to input and output dataclass's instead of dictionaries.

### CODE REVIEW

File: flatliners/comparisonscore.py [hunk 1]
```diff
@@ -53,16 +51,27 @@ def compute_cluster_distance(self, values):
...
         
           if resoure_name in self.clusters[cluster_id].keys():
               return True
           else:
               return False
 
+@dataclass
+class STD_COMPARISON_DATA:
+
+    cluster: str = ""
```
[the remark this answers is not in the document]
Comment (durandom) [reply flatliners/comparisonscore.py]: I think we should ignore E701 for those dataclasses:
Add this to .stickler.yml in this PR and we should be good

```diff
diff --git a/.stickler.yml b/.stickler.yml
index ae58708..167182d 100644
--- a/.stickler.yml
+++ b/.stickler.yml
@@ -3,9 +3,11 @@ linters:
     python: 3
     max-line-length: 160
     fixer: true
+    ignore: 'E701'
   pep8:
     python: 3
     max-line-length: 160
     fixer: true
+    ignore: 'E701'
 fixers:
   enable: true
```

File: flatliners/comparisonscore.py [hunk 2]
```diff
@@ -53,16 +51,27 @@ def compute_cluster_distance(self, values):
...
+          cluster_id = x.cluster
+          resoure_name = x.resource
         
           if resoure_name in self.clusters[cluster_id].keys():
               return True
           else:
               return False
 
+@dataclass
+class STD_COMPARISON_DATA:
```
Comment (durandom) [inline flatliners/comparisonscore.py]: why UPPER_CASE? 
And drop the `DATA` suffix.

**Review by durandom — COMMENTED** [review]
Maybe we'll nest the dataclasses **inside** the flatliner? This way we could always name them the same. E.g. `Data`, although `Data` is too generic, maybe you can come up with a better name? `Signal`, `Cue`, `Product`, `Result`?

Like this:

```python
class Human:

    def __init__(self):
        self.name = 'Guido'
        self.head = self.Head()

    class Head:
        def talk(self):
            return 'talking...'

if __name__ == '__main__':
    guido = Human()
    print (guido.name)
    print (guido.head.talk())
    head = Human.Head()
    print(head.talk())
```

this from https://pythonspot.com/inner-classes/

**Comment by MichaelClifford** [comment]
@durandom @4n4nd Please review this PR and let me know if you have any other comments. Thanks!

**Review by durandom — APPROVED** [review]
LGTM, but you need to rebase on top of master and run `pipenv install` again, to fix the conflicts in `Pipfile.lock`

File: metrics/__init__.py [hunk 3]
```diff
@@ -1,2 +1,3 @@
 from .filemetrics import FileMetrics
 from .prommetrics import PromMetrics
+from .prommetricslive import PromMetricsLive
```
[the remark this answers is not in the document]
Comment (4n4nd) [reply metrics/__init__.py:3]: Should I just comment this line?

[the remark this answers is not in the document]
Comment (durandom) [reply metrics/__init__.py:3]: I think we dont need to import anything here. maybe we can completly empty this file

**Pull request merged**